wagtail/documents/views/documents.py

```diff
@@ -106,21 +111,27 @@ def columns(self):
                 width="16%",
             ),
         ]
-        if self.collections:
+        if self.filters and "collection_id" in self.filters.filters:
             columns.insert(
                 3,
                 Column("collection", label=_("Collection"), accessor="collection.name"),
             )
         return columns
 
     @cached_property
-    def collections(self):
-        collections = permission_policy.collections_user_has_any_permission_for(
-            self.request.user, ["add", "change"]
-        )
-        if len(collections) < 2:
-            collections = None
-        return collections
+    def header_buttons(self):
+        buttons = []
+        if self.add_url:
+            buttons.append(
+                HeaderButton(
+                    self.add_item_label,
+                    url=self.add_url,
+                    icon_name="plus",
+                    attrs={"data-w-link-reflect-keys-value": '["collection_id"]'},
+                    icon_only=False,
+                )
+            )
+        return buttons
```

Comment:
As with #11698, this will not be necessary if #11727 lands before this PR (see also #11726).